core/src/main/java/org/ehcache/core/osgi/EhcacheActivator.java

```diff
@@ -17,23 +17,37 @@
 package org.ehcache.core.osgi;
 
 import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.ehcache.core.spi.service.ServiceFactory;
+import org.ehcache.core.util.ClassLoading;
 import org.osgi.framework.BundleActivator;
 import org.osgi.framework.BundleContext;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import static java.util.Spliterators.spliterator;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.StreamSupport.stream;
+
 public class EhcacheActivator implements BundleActivator {
 
+  public static final String OSGI_LOADING = "org.ehcache.core.osgi";
+
   private static final Logger LOGGER = LoggerFactory.getLogger(EhcacheActivator.class);
 
   private static volatile BundleContext CORE_BUNDLE;
 
   @Override
   @SuppressFBWarnings("ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD")
   public void start(BundleContext context) throws Exception {
-    LOGGER.info("Detected OSGi Environment (core is in bundle: " + context.getBundle() + "): Using OSGi Based Service Loading");
-    SafeOsgi.OSGI = true;
-    CORE_BUNDLE = context;
+    if ("true".equalsIgnoreCase(context.getProperty(OSGI_LOADING))) {
```

Comment:
I'm wondering about the default value for this. How easy it is to set the value for someone in OSGi? And how likely it is that they would want the OSGi service loader instead of the normal one?

Reply:
It can be either a system property or a framework property.  I believe there is no standardized way of setting framework properties... it's an implementation decision.  Apache Felix has a `conf/config.properties` file for example: http://felix.apache.org/documentation/subprojects/apache-felix-framework/apache-felix-framework-usage-documentation.html#configuring-bundles

Anyone using OSGi with transactions or clustered caching in OSS will either need to build their own bundle or use the OSGi service loading mechanism.  We could make this the default in OSGi, in principle it has no downside beyond "this is not battle hardened".

Reply:
I would put it the default then. Seems what is expected for a OSGi user